tweet_recommendations/rank_pipeline.py

```diff
@@ -1,20 +1,52 @@
-from tweet_recommendations.embeddings.fast_text import get_fasttext_tweet_embedding
-from tweet_recommendations.embeddings.word2vec import get_w2v_tweet_embedding
-from tweet_recommendations.utils.clients import get_wcrft2_results_for_text
+import random
+
+import pandas as pd
+import numpy as np
+
+from tweet_recommendations.utils.metrics import get_mean_average_precision_at_k, get_rank_dcg_at_k
+
+random.seed(1337)
 
 use_w2v_embedding = True
 
+data = pd.DataFrame(data={"hashtag": ["a", "b", "c", "d", "e", "f", "g", "h"]})
+config = {"hashtag_count": 3}
+testdict = {
+    "tweet": ["123", "456", "789"],
+    "expected_hashtags": [["a", "c", "g"], ["c", "d", "f"], ["a", "d", "h"]],
+}
+testset = pd.DataFrame(data=testdict)
+
 
-def get_hashtag_rank_for_given_tweet_text(tweet: str):
-    tagged_tweet_content = get_wcrft2_results_for_text(tweet)
+def get_k_random_hashtags_from_hashtag_data(hashtag_data, k=10):
+    hashtag_range = len(hashtag_data)
+    hashtag_list = []
+    for i in range(k):
+        hashtag_list.append(
+            hashtag_data.iloc[random.randint(0, hashtag_range - 1)]["hashtag"]
+        )
+    return hashtag_list
 
-    embeddings = (
-        get_w2v_tweet_embedding(tagged_tweet_content)
-        if use_w2v_embedding
-        else get_fasttext_tweet_embedding(tagged_tweet_content)
+
+def get_hashtag_rank_for_given_tweets(tweet, hashtag_data: pd.DataFrame, config: dict):
+    return get_k_random_hashtags_from_hashtag_data(
+        hashtag_data, config["hashtag_count"]
     )
 
-    hashtag_rank = []
-    # hashtag_rank = perform_some_graph_rank_magic(embeddings)
 
-    return hashtag_rank
+def get_map_value_for_tweet(test_set, hashtag_data=None):
```

Comment:
może to warto wrzucić do osobnego pliku, coś w stylu np. "testing_framework.py", a nie w kodzie rank_pipeline

Reply:
spoko, tak zrobię